src/dparse/lexer.d

```diff
@@ -1526,6 +1526,23 @@ private pure nothrow @safe:
             index);
     }
 
+    bool lexNamedEntity()
+    {
+        assert (range.bytes[range.index] == '&');
```

Comment:
I think it would be more D-style to write this kind of input as in contract. Because libdparse still supports older D versions in its travis file you need to do the full `in { assert(...); } do` code though.